pkg/semantic/version.go

```diff
@@ -1,19 +1,19 @@
 package semantic
 
 import (
-	"fmt"
 	"math/big"
-	"strings"
 )
 
-type Components []*big.Int
-
-type Version struct {
-	LeadingV   bool
-	Components Components
-	Build      string
+type Version interface {
+	// CompareStr returns an integer representing the sort order of the given string
+	// when parsed as the concrete Version relative to the subject Version.
+	//
+	// The result will be 0 if v == w, -1 if v < w, or +1 if v > w.
+	CompareStr(str string) int
```

Comment:
Should `Compare` also be part of the interface here?

Reply:
I found that having `Compare` would mean I'd need to handle the case where someone passes a `Version` that was for a different ecosystem, since the signature would have to be `Compare(version Version) (int, error)` - I wouldn't be surprised if that is useful in future, but for now by going with `CompareStr` I avoided having to handle that extra case

Reply:
ack, that makes sense!